src/vs/vscode.proposed.d.ts

```diff
@@ -416,6 +415,13 @@ declare module 'vscode' {
 		 * This extension's logger
 		 */
 		logger: Logger;
+
+		/**
+		 * Path where an extension can write log files.
+		 *
+		 * Extensions must create this directory before writing to it. The parent directory will always exist.
+		 */
+		readonly logDirectory: Uri;
```

Comment:
Unsure if we want this to be an `URI` or just a string, like `storagePath`. I guess it's unlikely that log files aren't stored under the `file`-scheme or that extensions are able to create a folder on the `foo`-scheme. 

So, I'd suggest to make this just a string or to clearly document that this is a `file`-scheme-uri.

Reply:
Yes, there's no real need for this to be a uri at the moment since this will always be a filesystem path but I do generally like using a distinct type for paths. Since we're asking users to use this path with the node fs apis, perhaps we should just use a string this time though